### ISSUE TITLE
CentOS detected as Amazon Linux

### ISSUE THREAD

**Issue by bcoles** [rapid7/metasploit-framework#10934]
The [get_sysinfo](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/post/linux/system.rb#L14) method in `Msf::Post::Linux::System` uses a massive `if/elsif` block to detect the Linux distribution. `/etc/system-release` is the second check, and was traditionally used to detect `amazon`.

```ruby
  #
  # Returns a Hash containing Distribution Name, Version and Kernel Information
  #
  def get_sysinfo
    system_data = {}
    etc_files = cmd_exec("ls /etc").split()

    kernel_version = cmd_exec("uname -a")
    system_data[:kernel] = kernel_version

    # Debian
    if etc_files.include?("debian_version")
      if kernel_version =~ /Ubuntu/
        version = read_file("/etc/issue").gsub(/\n|\\n|\\l/,'')
        system_data[:distro] = "ubuntu"
        system_data[:version] = version
      else
        version = read_file("/etc/issue").gsub(/\n|\\n|\\l/,'')
        system_data[:distro] = "debian"
        system_data[:version] = version
      end

    # Amazon
    elsif etc_files.include?("system-release")
      version = read_file("/etc/system-release").gsub(/\n|\\n|\\l/,'')
      system_data[:distro] = "amazon"
```

Some time around CentOS 6 or 7, it started shipping with a `/etc/system-release`. As such, CentOS is detected as Amazon.

**Comment by bcoles** [comment]
Fixed in #10927

**Comment by h00die** [comment]
merged
